### PR TITLE
Utilites -> internal submodule refactor

### PR DESCRIPTION
**Pull request by jwmueller**
Renames `Utilies` submodule to `internal`.  Updates docs to display methods previously missing such as those in `label_quality_utils.py`. Clarifies missing steps in instructions to build docs locally and adds more information to contributing guidelines (eg. how to run unit tests).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -29,23 +28,27 @@ appreciated!
...
+To contribute your code to the library, you'll want to create a new [Pull Request][pr]. 
 
-Cleanlab follows the [Black] style. Before you submit your pull request,
-install Black with `pip install black` and format the code with `black .`.
+Any changes to the code base should try to follow the style and coding conventions used in the rest of the project. Cleanlab follows the [Black] style. Before you submit your pull request, install Black with `pip install black` and format the code with `black`. New functions should be well-documented with accompanying unit tests that run quickly.
+
+Once you have finalized your edits to the cleanlab code, make sure the unit tests pass by executing:
+
+```
+pytest cleanlab/tests/
```
Comment (anishathalye) [inline CONTRIBUTING.md:47]: They can just do `pytest` in the repository root. In any case, the `cleanlab/` is confusing, I'd assume that the user is in the cleanlab (repo root) directory already, and `cleanlab/tests` does not exist.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -29,23 +28,27 @@ appreciated!
...
+```
+
+Also [build the docs from your local cleanlab version][instructions] and ensure the documentation for your new functions is formatted correctly and the existing functions' documentation remains valid. 
+Finally make sure all of the [examples] still work with the new code.
```
Comment (anishathalye) [inline CONTRIBUTING.md:51]: I don't think we should ask contributors to make sure all the examples work unless we (1) make it easy for them to run all the examples against their in-development version of cleanlab, (2) document how to do so, and (3) test this in CI for PRs.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -29,23 +28,27 @@ appreciated!
...
-install Black with `pip install black` and format the code with `black .`.
+Any changes to the code base should try to follow the style and coding conventions used in the rest of the project. Cleanlab follows the [Black] style. Before you submit your pull request, install Black with `pip install black` and format the code with `black`. New functions should be well-documented with accompanying unit tests that run quickly.
+
+Once you have finalized your edits to the cleanlab code, make sure the unit tests pass by executing:
+
+```
+pytest cleanlab/tests/
+```
+
+Also [build the docs from your local cleanlab version][instructions] and ensure the documentation for your new functions is formatted correctly and the existing functions' documentation remains valid.
```
Comment (anishathalye) [inline CONTRIBUTING.md:50]: This also adds friction for contributors. Until it's automated, maybe we the maintainers should just handle this aspect.

**Review by anishathalye — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (jwmueller) [reply CONTRIBUTING.md:47]: edited to: 

Once you have finalized your edits to the cleanlab code, make sure the unit tests pass by executing: `pytest` from the root cleanlab/ directory.

File: CONTRIBUTING.md (hunk 3 above)
Comment (jwmueller) [reply CONTRIBUTING.md:50]: Changed to optional language below. I think it's helpful to at least point out how to do this.


You can optionally [build the docs from your local cleanlab version][instructions] to check that the documentation for your new functions is formatted correctly and the existing functions' documentation remains valid.

**Pull request merged**